### PR TITLE
Add basic serialization to JSON for Result

### LINKED ISSUE #42: Allow Filing to be serialized to JSON
We need to be able to serialize `Filing` to JSON either individually or as a bundle so that modules can make use of it.

### PR DESCRIPTION
**Pull request by glesica**
Closes #42 

We want to be able to serialize a set of `Filing`s to JSON so that they can be used by the client modules. Add a basic capability to do that.

**Merging into my other branch right now to keep the diff clean. #41 should merge first and then this can be re-targeted to master.**

### CODE REVIEW

File: extractor/extractor/result.py [hunk 1]
```diff
@@ -72,3 +72,13 @@ def skip(self, n: int) -> Result:
...
         pass
+
+    def to_json(self) -> Dict[str, Any]:
+        """
+        Convert the result into JSON suitable for use as data for modules.
+
+        TODO: Think about additional metadata we might want
+        """
+        return {
+            "filings": [f.to_json() for f in self],
```
Comment (smai-f) [inline extractor/extractor/result.py:83]: https://github.com/code-for-montana/nonprofit-data/pull/43/files#diff-e68e2639de57d960287faf4715b35682R64

Probably missing something but how are the quotation marks being added around the values here ^^? I would have kinda expected needing to do a `json.dumps` here as well?

**Review by smai-f — APPROVED** [review]
assuming my comment is a dumb one, looks good to me

File: extractor/extractor/result.py (hunk 1 above)
Comment (glesica) [reply extractor/extractor/result.py:83]: Here we just build a dictionary that contains a single key, mapped to a list of other dictionaries, and dump it to JSON in the formatter (or wherever). It's basically the same as how our `toJson` methods in Dart return a `Map`.

Comment (smai-f) [reply extractor/extractor/result.py:83]: For posterity: the real JSON encoding happens in `formatter.py` which is added as a follow-up to this PR. In this PR the app `dump` is just what's used as proof of concept.

**Comment by smai-f** [comment]
QA +1

```
{"filings": [{"activity_or_mission_description": "DISTRIBUTION OF LITERATURE, MUSIC, AND OTHER RELATED RESOURCES WHICH COMPLIMENT LITERATURE; SUPPORT OF MINISTRIES.", "formation_year": "1981", "gross_receipts": 130306, "employee_count": 0, "principal_officer_name": "MAUREEN CARVILLE", "us_address": "55 CARTER STREET", "us_city_name": "BERLIN", "us_zip_code": "01503", "website_address": null}, {"activity_or_mission_description": "THE OBJECTIVE OF THIS ORGANIZATION SHALL BE TO PERPETUATE THE NAMES AND ACHIEVEMENTS OF THE MEN, WOMEN AND CHILDREN WHO WERE THE PIONEERS IN FOUNDING THIS COMMONWEALTH: BY PRESERVING OLD LANDMARKS, MARKING HISTORICAL PLACES, COLLECTING ARTIFACTS AND HISTORIES, ESTABLISHING A LIBRARY OF HISTORICAL MATTER AND SECURING MANUSCRIPTS, PHOTOGRAPHS, MAPS AND ALL SUCH DATA AS SHALL AID IN PERFECTING A RECORD OF THE UTAH PIONEERS; BY COMMEMORATING THEIR ENTRANCE INTO THE VALLEY OF THE GREAT SALT LAKE ON JULY 24, 1847, AND SUCH OTHER EVENTS AND DAYS AS ARE IMPORTANT IN THE EARLY HISTORY OF THIS COMMUNITY; BY PUBLISHING HISTORICAL MATERIAL; BY REVIEWING THE LIVES OF THE PIONEERS; THUS TEACHING THEIR DESCENDANTS AND THE CITIZENS OF OUR COUNTRY LESSONS OF FAITH, COURAGE, FORTITUDE, AND PATRIOTISM. THIS ORGANIZATION IS INTERNATIONAL IN ITS SCOPE AND IS ORGANIZED SOLELY FOR HISTORICAL, EDUCATIONAL, AND PUBLIC PURPOSES; IT IS NON-POLITICAL AND NON- SECTARIAN.", "formation_year": "1925", "gross_receipts": 510313, "employee_count": 14, "principal_officer_name": "CHERYL SEARLE", "us_address": "300 N MAIN ST", "us_city_name": "SALT LAKE CITY", "us_zip_code": "84103", "website_address": "WWW.DUPINTERNATIONAL.ORG"}, {"activity_or_mission_description": "TO SERVE CLASSROOMS IN NEED BY OPERATING A VOLUNTEER-RUN FREE STORE FOR TEACHERS.SINCE OPENING OUR DOORS IN 1999, OUR MISSION REMAINS THE SAME: TO SUPPORT PUBLIC EDUCATION IN THE PORTLAND AREA BY PROVIDING FREE SCHOOL SUPPLIES TO STUDENTS IN NEED. WE DO THIS BECAUSE WE BELIEVE THAT EVERY CHILD DESERVES SCHOOL SUPPLIES AND HAS THE RIGHT TO A QUALITY EDUCATION REGARDLESS OF THEIR FAMILY'S INCOME OR RACIAL/ETHNIC IDENTITY. OUR ULTIMATE GOAL IS TO LEVEL THE PLAYING FIELD SO EVERY CHILD HAS ACCESS TO THE TOOLS THAT ALLOW THEM TO ACHIEVE ACADEMIC SUCCESS AND FEEL CONFIDENT. THE PROGRAM DELIVERY MODEL ACHIEVES ADDITIONAL GOALS THROUGH COMMUNITY ENGAGEMENT AND BY KEEPING SUPPLIES OUT OF THE WASTE STREAM. WITH THE COMMUNITY'S SUPPORT, SCHOOLHOUSE SUPPLIES HAS DISTRIBUTED NEARLY $30 MILLION IN FREE SCHOOL SUPPLIES AND BENEFITTED HUNDREDS OF THOUSANDS OF CHILDREN.", "formation_year": "2006", "gross_receipts": 4058503, "employee_count": 10, "principal_officer_name": "ALICE FORBES", "us_address": "4916 NE 122ND AVENUE", "us_city_name": "PORTLAND", "us_zip_code": "97230", "website_address": "WWW.SCHOOLHOUSESUPPLIES.ORG"}, {"activity_or_mission_description": "MINDFUL LIFE PROJECT'S MISSION IS TO EMPOWER UNDERSERVED CHILDREN THROUGH MINDFULNESS AND OTHER TRANSFORMATIVE SKILLS TO GAIN SELF-AWARENESS, CONFIDENCE, SELF-REGULATION AND RESILIENCE, LEADING TO LIFELONG SUCCESS. MINDFUL LIFE PROJECT (MLP) ACHIEVES THIS BY OFFERING DIRECT SERVICE PROGRAMS IN SCHOOLS: MINDFUL COMMUNITY AND RISE-UP.", "formation_year": "2015", "gross_receipts": 1015970, "employee_count": 27, "principal_officer_name": "JG LAROCHETTE", "us_address": "124 WASHINGTON AVE STE B", "us_city_name": "RICHMOND", "us_zip_code": "94801", "website_address": "WWW.MINDFULLIFEPROJECT.ORG"}]}
```

**Pull request merged**